### DISCUSSION TITLE
Having trouble with ESP32-S3-WROOM-1 and gpio config

### DISCUSSION THREAD

**Discussion by jdeltoft** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#573] [Q&A] [answered]
Starting a new post for this ESP32-S3 question.  I have had my project working well with the "default" ESP32 now for a bit, but wanted more RAM.  So I ordered some S3s with 8MB RAM, 16MB flash.  Took a tiny bit of work to get them flashing but wasn't too bad.  I then figured it wouldn't be too hard to move over the 14 wires to the 16 pin HUB75E header.   Alas...  :-(

So after flashing, wifi and logging look perfect, as with other ESP32.  But when I moved the GPIO pins to the new board, it's flickering lines on the led panels.  I actually had something like this a long time ago with esp32 but it was just a swapped line. So I didn't panic, just checked my header solder joints, re-connected being extra careful for pin numbers etc.  No luck.

(There is good news also, the free heap is `Free HEAP: 8479959`)

Here is my new config in code (src/main.h):

```
//////////////////// side by notch
  #define R1    6
  #define BL1  17
  #define R2   16
  #define BL2  14
  #define CH_A 13
  #define CH_C 11
  #define CLK   5
  #define OE    4
//////////////////// side away from notch
  #define G1    7
// Ground Pin
  #define G2   15
  #define CH_E 10
  #define CH_B 12
  #define CH_D 18
  #define LAT   9
// Ground Pin
////////////////////

```
And here is the board [gpio](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/gpio.html) numbering:

<img width="610" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/d4debf2c-e29f-4aaf-829d-4a0f1e478688">

I looked at the esp32 page to see if any of those pins was a bad choice but don't see any issue.  Is there something I'm missing? 

Any ideas what to try to debug this?

Here is the board look:

<img width="950" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/74c74a21-266b-43fb-b31d-606116e6f2a9">

**Comment by jdeltoft** [comment]
Per a previous discussion, I went back to https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/blob/master/doc/BuildOptions.md For setting up the S3 

I turned on 

```
build_flags = -DSPIRAM_FRAMEBUFFER -DCORE_DEBUG_LEVEL=3
```
and learned that my configuration for gpio is not working? I will walk back to the previous non S3 working setup for more clues.

<img width="853" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/7f74ef28-57e1-4df6-bcec-003ef5d31ec1">

**Comment by jdeltoft** [comment]
Ok, looks like this main.h set of defines is not actually used.  I went back to the non S3 board and forgot to change the `#if 0` but let it run anyway and the gpio changed still.  

```
[   663][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:415] begin(): [begin()] Using GPIO 25 for R1_PIN
[   664][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:416] begin(): [begin()] Using GPIO 26 for G1_PIN
[   672][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:417] begin(): [begin()] Using GPIO 27 for B1_PIN
[   680][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:418] begin(): [begin()] Using GPIO 14 for R2_PIN
[   689][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:419] begin(): [begin()] Using GPIO 12 for G2_PIN
[   697][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:420] begin(): [begin()] Using GPIO 13 for B2_PIN
[   705][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:421] begin(): [begin()] Using GPIO 23 for A_PIN
[   713][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:422] begin(): [begin()] Using GPIO 19 for B_PIN
[   722][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:423] begin(): [begin()] Using GPIO 5 for C_PIN
[   730][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:424] begin(): [begin()] Using GPIO 17 for D_PIN
[   738][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:425] begin(): [begin()] Using GPIO 10 for E_PIN
[   746][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:426] begin(): [begin()] Using GPIO 4 for LAT_PIN
[   755][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:427] begin(): [begin()] Using GPIO 15 for OE_PIN
[   763][I][ESP32-HUB75-MatrixPanel-I2S-DMA.h:428] begin(): [begin()] Using GPIO 16 for CLK_PIN
```
The other change I am doing is the `board` in platformio.ini.

<img width="500" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/7f8f2c4d-e070-4e64-b770-6454c8dcadfb">

Will be a good chance to go learn more about that board definition and how it's setting these gpio values.

**Comment by jdeltoft** [comment, marked as answer]
Ok, that works now if I force the file

`./libdeps/esp32/ESP32 HUB75 LED MATRIX PANEL DMA Display/src/platforms/esp32s3/esp32s3-default-pins.hp`

Just confused why the values in main.h don't work still.

Somewhere I lost a 90 degree rotation also.  😃 

<img width="964" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/366501f5-5ccb-4a9d-af98-6eed21571f5b">

**Comment by jdeltoft** [comment]
Glad these pins can work also since I sent this out yesterday...  hahaha

<img width="418" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/9b032e8e-ee09-4acd-b0fb-19c1c43ef9b0">

**Comment by jdeltoft** [comment]
<img width="752" alt="image" src="https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/4506303/98d2cd4f-2e2d-4db6-959b-def30670fd88">

This fixed it also.

**Reply by mrcodetastic** [reply]
This is indeed that is how gpio mapping is done. 

Would you mind sharing your pcb sketch? Does this fit just the standard s3 dev board? I would like to use this design in something I'm working on.

**Comment by jdeltoft** [comment]
Of course! I should get the board by next weekend, I would like to validate the board before sharing it here but as soon as I do I'll share the zip and also I can share links to the sd card module (spi) that I bought as an option (pins in the center) and the 16 pin through hole connecter also.  I can send pix also.  NOTE: This is specific to the 44 (22 per side) esp32 S3 boards I got, but the footprint is pretty standard (.1'' spacing, 1'' spacing between two sides)

Regarding these pins, I am still confused (haven't thought about it much) why just setting up main.h values doesn't work?  Can you share why it was setup that way?
